embedded-io-async/src/impls/vec_deque.rs

```diff
@@ -0,0 +1,83 @@
+//! Adapted from std.
```

Comment:
It's pretty verbatim in places. I think that's fine given that both are licensed under "MIT OR Apache-2.0", and the original files do not have any copyright notes that point out concrete contributors (so the embedded-hal copyright notice should already cover that).